homeassistant/components/solaredge/__init__.py

```diff
@@ -1,18 +1,50 @@
-"""The solaredge integration."""
+"""The SolarEdge integration."""
 from __future__ import annotations
 
+from requests.exceptions import ConnectTimeout, HTTPError
+from solaredge import Solaredge
+
 from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_API_KEY
 from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
 import homeassistant.helpers.config_validation as cv
 
-from .const import DOMAIN
+from .const import CONF_SITE_ID, DATA_API_CLIENT, DOMAIN, LOGGER
 
 CONFIG_SCHEMA = cv.deprecated(DOMAIN)
 
+PLATFORMS = ["sensor"]
+
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
-    """Load the saved entities."""
-    hass.async_create_task(
-        hass.config_entries.async_forward_entry_setup(entry, "sensor")
-    )
+    """Set up SolarEdge from a config entry."""
+    api = Solaredge(entry.data[CONF_API_KEY])
+
+    try:
+        response = await hass.async_add_executor_job(
+            api.get_details, entry.data[CONF_SITE_ID]
+        )
+    except KeyError as ex:
+        LOGGER.error("Missing details data in SolarEdge response")
+        raise ConfigEntryNotReady from ex
+    except (ConnectTimeout, HTTPError) as ex:
+        LOGGER.error("Could not retrieve details from SolarEdge API")
+        raise ConfigEntryNotReady from ex
+
+    if response["details"]["status"].lower() != "active":
```

Comment:
I expect the KeyError exception should have catched missing keys in the the `response` dict, not the `entry.data` dict. @frenck Maybe the slim-down of the try-except block should be reverted.

Reply:
More refactoring on the way. Its ok.